backend/cn/lib/report.mli

```diff
@@ -23,21 +19,36 @@ type where_report =
   { fnction : string option;
     section : string option;
     loc_cartesian : ((int * int) * (int * int)) option;
-    loc_head : string
+    (** Where in the source file we are *)
+    loc_head : string (** Name of what we are currently processing *)
   }
 
+(** Information about a specific state of the computation.
+    The resources, constraints, and terms are pairs because they classify
+    how relevant the thing might be:
+    the first component is "interesting", the second is not. *)
 type state_report =
-  { where : where_report;
-    resources : Pp.document list * Pp.document list;
-    constraints : Pp.document list * Pp.document list;
-    terms : term_entry list * term_entry list
+  { where : where_report; (** Location information *)
+    resources : Pp.document list * Pp.document list; (** Resources *)
+    constraints : Pp.document list * Pp.document list; (** Constraints *)
+    terms : term_entry list * term_entry list (** Term values *)
```

Comment:
Just for future reference, my preference is that comments should be there to explain _why_ rather than what and variable names should be self explanatory as far as possible. If the variable name would end up being too long, then a short name plus a comment (e.g. as in lines 39-43) is good.